src/Microsoft.Extensions.DependencyModel/DependencyContext.cs

```diff
@@ -15,13 +15,46 @@ public class DependencyContext
 
         private static readonly Lazy<DependencyContext> _defaultContext = new Lazy<DependencyContext>(LoadDefault);
 
-        public DependencyContext(string target, string runtime, CompilationOptions compilationOptions, CompilationLibrary[] compileLibraries, RuntimeLibrary[] runtimeLibraries)
+        public DependencyContext(string target,
+            string runtime,
+            bool isPortable,
+            CompilationOptions compilationOptions,
+            CompilationLibrary[] compileLibraries,
+            RuntimeLibrary[] runtimeLibraries,
+            IReadOnlyList<KeyValuePair<string, string[]>> runtimeGraph)
         {
+            if (target == null)
+            {
+                throw new ArgumentNullException(nameof(target));
+            }
+            if (runtime == null)
+            {
+                throw new ArgumentNullException(nameof(runtime));
+            }
+            if (compilationOptions == null)
+            {
+                throw new ArgumentNullException(nameof(compilationOptions));
+            }
+            if (compileLibraries == null)
+            {
+                throw new ArgumentNullException(nameof(compileLibraries));
+            }
+            if (runtimeLibraries == null)
+            {
+                throw new ArgumentNullException(nameof(runtimeLibraries));
+            }
+            if (runtimeGraph == null)
+            {
+                throw new ArgumentNullException(nameof(runtimeGraph));
+            }
+
             Target = target;
```

Comment:
I know this isn't new, but this should be `TargetFramework`. I feel like `Target` is too generic.